### ISSUE TITLE
question about kernel name

### ISSUE THREAD

**Issue by bader** [intel/llvm#4569]
### Discussed in https://github.com/intel/llvm/discussions/4551

<div type='discussions-op-text'>

<sup>Originally posted by **zjin-lcf** September 13, 2021</sup>
When the kernel name is "distance" (i.e. <class distance>), the compiler error implies that the name is in conflict with something defined in sycl_span.hpp. 

The compiler reported the error with the HIP/ROCm backend. The compiler does not report the error for the CUDA backend. Could you reproduce the error using any sycl program ?

```
error: invalid operands to binary expression ('size_t' (aka 'unsigned long') and 'distance')
    _SYCL_SPAN_ASSERT(_Extent == distance(__f, __l),
``` </div>
